src/data/list/basic.lean

```diff
@@ -4658,4 +4658,26 @@ begin
       apply lt_of_succ_lt_succ hi, } },
 end
 
+/-- Backwards recursor on lists, inducts by appending elements to the end of the list -/
+@[elab_as_eliminator]
+def backwards_rec {α : Type u} {P : list α → Sort v} (hPempty : P [])
+  (hPind : ∀ l x, P l → P (l ++ [x])) : Π l, P l :=
+begin
+  suffices : Π l : list α, P l.reverse,
+  { intro l,
+    convert this l.reverse,
+    exact l.reverse_reverse.symm },
+  intro l,
+  induction l with h t ih,
+  { exact hPempty },
+  { rw list.reverse_cons,
+    apply hPind,
+    exact ih }
+end
```

Comment:
Golfs to:
```lean
l.reverse_reverse ▸
  list.rec hPempty (λ h t, (list.reverse_cons h t).symm ▸ hPind _ _) l.reverse
```
will leave it to reviewers to decide if that's a good idea

Reply:
I'm not a big fan of ▸, I found it almost unreadable and also very brittle when you change something.  My preference is always to use rw explicitly:
```lean
begin
  rw ← reverse_reverse l,
  induction l.reverse,
  { exact hPempty },
  { rw reverse_cons, exact hPind _ _ ih },
end
```